elements/bulbs-reading-list/components/bulbs-reading-list-item.js

```diff
@@ -3,14 +3,146 @@ import {
   BulbsHTMLElement,
 } from 'bulbs-elements/register';
 import invariant from 'invariant';
+import {
+  debouncePerFrame,
+  filterBadResponse,
+  getResponseText,
+  getAnalyticsManager,
+  InViewMonitor,
+  prepGaEventTracker,
+} from 'bulbs-elements/util';
+
+let pageStartDebouncer = debouncePerFrame();
 
 class BulbsReadingListItem extends BulbsHTMLElement {
   attachedCallback () {
-    invariant(this.dataset.id, '<bulbs-reading-list-menu> requires a "data-id" attribute');
-    invariant(this.dataset.href, '<bulbs-reading-list-menu> requires a "data-href" attribute');
-    invariant(this.dataset.partialUrl, '<bulbs-reading-list-menu> requires a "data-partial-url" attribute');
-    invariant(this.dataset.title, '<bulbs-reading-list-menu> requires a "data-title" attribute');
+    const markdownText = '<bulbs-reading-list-item> requires attribute: ';
+    invariant(this.dataset.id, markdownText + 'data-id');
+    invariant(this.dataset.href, markdownText + 'data-href');
+    invariant(this.dataset.partialUrl, markdownText + 'data-partial-url');
+    invariant(this.dataset.title, markdownText + 'data-title');
+    invariant(this.dataset.contentAnalyticsDimensions,
+      markdownText + 'data-content-analytics-dimensions');
+
+    InViewMonitor.add(this);
+    this.id = parseInt(this.dataset.id, 10);
+    this.href = this.dataset.href;
+    this.partialUrl = this.dataset.partialUrl;
+    this.title = this.dataset.title;
+    this.gaDimensions = this.getGaDimensions();
+
+    this.isLoaded = false;
+    this.fetchPending = false;
+    this.loadingTemplate = '<p><i class="fa fa-spinner fa-spin"></i> Loading...</p>';
+
+    this.registerEvents();
+  }
+
+  registerEvents () {
+    this.addEventListener('approachingviewport', this.loadContent.bind(this));
+    this.addEventListener('pagestart', this.handlePageStart.bind(this));
+  }
+
+  getGaDimensions () {
+    let targeting = JSON.parse(
+      this.dataset.contentAnalyticsDimensions
+    );
+    return {
+      'dimension1': targeting.dimension1 || 'None',
+      'dimension2': targeting.dimension2 || 'None',
+      'dimension3': targeting.dimension3 || 'None',
+      'dimension4': targeting.dimension4 || 'None',
+      'dimension5': targeting.dimension5 || 'None',
+      'dimension6': targeting.dimension6 || 'None',
+      'dimension7': targeting.dimension7 || 'None',
+      'dimension8': targeting.dimension8 || 'None',
+      'dimension9': targeting.dimension9 || 'None',
+      'dimension10': targeting.dimension10 || 'None',
+      'dimension11': targeting.dimension11 || 'None',
```

Comment:
Reminder to chat with @jmelvnsn to pull this in along the way (https://github.com/theonion/bulbs-elements/pull/270)